pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        transfer_function_response_string = self._dll_interface.get_string(
+            self._dll.getIdealTransferFunctionResponse, max_size=size.value
+        )
+        return transfer_function_response_string
+
+    @property
+    def vsg_analysis_enabled(self) -> bool:
+        """Flag indicating if the offset due to source resistor in frequency and time responses is enabled.
+
+        Returns
+        -------
+        bool
+        """
+
+        vsg_analysis_enabled = c_bool()
+        status = self._dll.getVSGAnalsyis(byref(vsg_analysis_enabled))
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+        return bool(vsg_analysis_enabled.value)
+
+    @vsg_analysis_enabled.setter
+    def vsg_analysis_enabled(self, filter_vsg_analysis_enabled: bool):
+        status = self._dll.setVSGAnalsyis(filter_vsg_analysis_enabled)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def frequency_response(
+        self,
+        y_axis_parameter=FrequencyResponseColumn.MAGNITUDE_DB,
+        minimum_frequency="200 MHz",
+        maximum_frequency="5 GHz",
+        vsg_analysis_enabled=False,
+    ):
+        """Return the ideal filter frequency response for the given parameters.
+
+        Parameters
+        ----------
+        y_axis_parameter: `FrequencyResponseColumn`, optional
+            Parameter to return. The default is frequency response magnitude in dB.
+        minimum_frequency: str, optional
+            The default is ``200 MHz``.
+        maximum_frequency: str, optional
+            The default is ``5 GHz``.
+        vsg_analysis_enabled: bool, optional
+            The default is ``False``.
+
+        Returns
+        -------
+        tuple: The tuple contains
+                list of str:
+                    The defined frequency range.
+                list of str:
+                    The requested parameter.
+        """
+        self.graph_setup.minimum_frequency = minimum_frequency
+        self.graph_setup.maximum_frequency = maximum_frequency
+        self.vsg_analysis_enabled = vsg_analysis_enabled
+        frequency = self._frequency_response_getter(FrequencyResponseColumn.FREQUENCY)
+        frequency_hz = []
+        for freq in frequency:
+            frequency_hz.append(freq / (2 * math.pi))
+        parameter = self._frequency_response_getter(y_axis_parameter)
+        return frequency_hz, parameter
+
+    def time_response(
+        self,
+        y_axis_parameter=TimeResponseColumn.STEP_RESPONSE,
+        minimum_time="0 s",
+        maximum_time="10 ns",
+        vsg_analysis_enabled=False,
+    ):
+        """Return the ideal filter time response for the given parameters.
+
+        Parameters
+        ----------
+        y_axis_parameter: `TimeResponseColumn`, optional
+            Parameter to return. The default is step time response.
```

Comment:
```suggestion
            Time response column to get. The default is the step time response.
```
The rest of the parameters need to have descriptions added.